AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/hierarchical-state-machine-swift/StateMachine.swift

```diff
@@ -146,43 +111,23 @@ class StateMachine<
     /// and the time the pending cancellation is processed, the event will not be dispatched to the listener.
     ///
     /// - Parameter listenerToken: Identifies the listener to be removed
-    func cancel(listenerToken: StateChangeListenerToken) {
+    nonisolated func cancel(listenerToken: StateChangeListenerToken) {
         pendingCancellations.with { $0.insert(listenerToken) }
-        operationQueue.addOperation {
-            self.removeSubscription(listenerToken: listenerToken)
+        Task {
+           await self.removeSubscription(listenerToken: listenerToken)
         }
-    }
 
-    /// Invokes `completion` with the current state
-    ///
-    /// - Parameter completion: a callback to invoke with the current state
-    func getCurrentState(_ completion: @escaping (StateType) -> Void) {
-        operationQueue.addOperation {
-            completion(self.currentState)
-        }
-    }
-
-    var currentMachineState: StateType {
-        get async {
-            await withCheckedContinuation { continuation in
-                self.operationQueue.addOperation {
-                    continuation.resume(returning: self.currentState)
-                }
-            }
-        }
     }
 
     // MARK: - Isolated methods
 
-    /// Must be invoked on operationQueue
+    ///  Add the token in a weakly retained map
```

Comment:
suggestion: 
```suggestion
    ///  Weakly retain the listener and synchronously invoke it with the current state
```

See note below about sync invocation